app/src/main/java/com/practicum/playlistmaker/Resource.kt

```diff
@@ -0,0 +1,6 @@
+package com.practicum.playlistmaker
+
+sealed class Resource<T> (val data: T? = null, val message: String? = null, val code: Int){
+    class Success<T>(data: T, code: Int): Resource<T>(data, code = code)
+    class Error<T>(message: String, data: T? = null, code: Int): Resource<T>(data, message, code)
```

Comment:
```suggestion
    class Error<T>(data: T? = null, message: String, code: Int): Resource<T>(data, message, code)
```
Писать нужно в том же порядке, что и параметры sealed class, или используй разный порядок и именованные переменные